dbms/src/Functions/FunctionsStringSearch.h

```diff
@@ -52,9 +52,11 @@ class IlikeLowerHelper
             break;
         case TiDB::ITiDBCollator::CollatorType::UTF8MB4_GENERAL_CI:
         case TiDB::ITiDBCollator::CollatorType::UTF8MB4_UNICODE_CI:
-        case TiDB::ITiDBCollator::CollatorType::UTF8MB4_0900_AI_CI:
             collator = TiDB::ITiDBCollator::getCollator(TiDB::ITiDBCollator::UTF8MB4_BIN);
             break;
+        case TiDB::ITiDBCollator::CollatorType::UTF8MB4_0900_AI_CI:
```

Comment:
It's a little strange here to move the UTF8MB4_0900_AI_CI case while leave 0900_BIN not handled, please double check it.

Reply:
I think it means the bin collator if `UTF8MB4_0900_AI_CI` is `UTF8MB4_0900_BIN`?

Reply:
Yes. The bin collator of `UTF8MB4_0900_AI_CI` is `UTF8MB4_0900_BIN`.

Actually, the pattern match behavior of `UTF8MB4_0900_BIN` and `UTF8MB4_BIN` is the same (and I think this function is only used in `ilike`). This change doesn't affect the behavior. I made this change just because it's more intuitive to set the binary collator of `UTF8MB4_0900_AI_CI` to be `UTF8MB4_0900_BIN`.